test/Transforms/BasisConversion/ctrl-s-cx.qke

```diff
@@ -0,0 +1,36 @@
+// ========================================================================== //
```

Comment:
What do you think about merging these basis conversions tests into one test file? It doesn't seem necessary to have each one in a separate file to me. (I know that's the existing pattern here, but it seems like we might risk duplicating tests with the combinatorics and having so many files.)

Reply:
I think it's the right thing to do. The difficulty is that each file may have a different `RUN` command (different pass pipelines).
I've merged the two basic test files (with `CircuitCheck`) in this PR into one. The one with the complex circuit in the bug report cannot be checked by `CircuitCheck` hence still needs to be in a separate file.